build-tools/automation/azure-pipelines-nightly.yaml

```diff
@@ -58,6 +68,173 @@ stages:
         artifactName: Build Results - Nightly macOS
         includeBuildResults: true
 
+- stage: designer_tests
+  displayName: Designer Tests
+  dependsOn: mac_build
+  variables:
+    - group: Xamarin-Secrets
+    - group: xamops-azdev-secrets
+  jobs:
+  # Check - "Xamarin.Android (macOS > Tests > Designer Integration)"
+  - job: designer_integration_mac
+    condition: false #TODO: Enable once test issues are fixed.
+    displayName: macOS > Tests > Designer Integration
+    pool:
+      vmImage: $(HostedMacImage)
+    timeoutInMinutes: 120
+    cancelTimeoutInMinutes: 5
+    workspace:
+      clean: all
+    variables:
+      EnableRegressionTest: true
+    steps:
+    - checkout: uitools
+      clean: true
+      submodules: recursive
+      path: s/UITools
+      persistCredentials: true
+
+    - powershell: |
+        # Use the branch name of the source being built or the PR target branch name. Fall back to 'main' if the branch is unknown.
+        $branchPrefix = "/refs/heads/"
+        $branchName = "$(Build.SourceBranch)" -replace $branchPrefix, ""
+        if ("$(Build.Reason)" -eq "PullRequest") {
+            $branchName = "$(System.PullRequest.TargetBranch)" -replace $branchPrefix, ""
+        }
+        if (("$branchName" -ne "main") -and ("$branchName" -notlike "d1*")) {
+            $branchName = "main"
+        }
+        Set-Location -Path $(System.DefaultWorkingDirectory)/UITools
+        git checkout $branchName
+        git submodule update -q --init --recursive
+      displayName: Clone and update UITools
```

Comment:
This step would have us checkout the d17-5 branch of UITools, but I think we'd want to track their main against our d17-5? I think we could probably remove this check.